components/dashboard/src/components/UsageView.tsx

```diff
@@ -159,261 +135,309 @@ function UsageView({ attributionId }: UsageViewProps) {
+                    dateFormat={"MMM d, yyyy"}
+                    // tab loop enabled causes a bug w/ layout shift to the right of input when open
+                    enableTabLoop={false}
+                />
+                <Subheading>to</Subheading>
+                <DatePicker
+                    selected={endDate.toDate()}
+                    onChange={(date) => date && onEndDateChange(dayjs(date))}
+                    selectsEnd
+                    startDate={startDate.toDate()}
+                    endDate={endDate.toDate()}
+                    minDate={startDate.toDate()}
+                    customInput={<DateDisplay />}
+                    dateFormat={"MMM d, yyyy"}
+                    enableTabLoop={false}
+                />
+            </div>
+        </div>
+    );
+};
+
+const DateDisplay = forwardRef((arg: any, ref: any) => (
+    <div
+        className="px-2 py-0.5 text-gray-500 bg-gray-50 dark:text-gray-400 dark:bg-gray-800 rounded-md cursor-pointer flex items-center hover:bg-gray-100 dark:hover:bg-gray-700"
+        onClick={arg.onClick}
+        ref={ref}
+    >
+        <div className="w-28 font-medium">{arg.value}</div>
+        <div>
+            <svg
+                width="20"
+                height="20"
+                fill="currentColor"
+                xmlns="http://www.w3.org/2000/svg"
+                onClick={arg.onClick}
+                ref={ref}
+            >
+                <path
+                    fillRule="evenodd"
+                    clipRule="evenodd"
+                    d="M5.293 7.293a1 1 0 0 1 1.414 0L10 10.586l3.293-3.293a1 1 0 1 1 1.414 1.414l-4 4a1 1 0 0 1-1.414 0l-4-4a1 1 0 0 1 0-1.414Z"
+                />
+                <title>Change Date</title>
+            </svg>
+        </div>
+    </div>
+));
+
+type UsageDateRangePickerProps = {
+    onChange: (start: dayjs.Dayjs, end: dayjs.Dayjs) => void;
+};
+const UsageDateRangePicker: FC<UsageDateRangePickerProps> = ({ onChange }) => {
+    const entries = useMemo<ContextMenuEntry[]>(() => {
+        const startOfCurrentMonth = dayjs().startOf("month");
+
+        const entries: ContextMenuEntry[] = [
+            {
+                title: "Current month",
+                onClick: () => onChange(startOfCurrentMonth, dayjs()),
+                active: false,
+            },
+        ];
+
+        // This goes back 6 months from the current month
+        for (let i = 1; i < 7; i++) {
+            const startDate = dayjs().subtract(i, "month").startOf("month");
+            const endDate = startDate.endOf("month");
+            entries.push({
+                title: startDate.format("MMMM YYYY"),
+                active: false,
+                onClick: () => onChange(startDate, endDate),
+            });
+        }
+
+        return entries;
+    }, [onChange]);
+
+    return (
+        <ContextMenu menuEntries={entries} customClasses="left-0">
+            <DateDisplay value="Date Range" onClick={noop} />
```

Comment:
**thought**: Now that we moved these out of the header we could fall back to regular dropdown inputs for better accessibility, but maybe this is more effort than I can estimate since the range picker is a weird library we're using.

Reply:
Something like this maybe?

| Light | Dark |
|--------|--------|
| ![image](https://github.com/gitpod-io/gitpod/assets/367275/2b702be7-25b1-4ea6-9f65-ae430925abdf) | ![image](https://github.com/gitpod-io/gitpod/assets/367275/e5c20458-4c3b-4531-928d-ac06d7db20cc) |